src/parsing/compileAstToExpression.ts

```diff
@@ -866,65 +870,71 @@ function pathExpr(ast: IAST, compilationOptions: CompilationOptions) {
 
 		if (axis) {
 			hasAxisStep = true;
-			const test = astHelper.getFirstChild(step, [
-				'attributeTest',
-				'anyElementTest',
-				'piTest',
-				'documentTest',
-				'elementTest',
-				'commentTest',
-				'namespaceTest',
-				'anyKindTest',
-				'textTest',
-				'anyFunctionTest',
-				'typedFunctionTest',
-				'schemaAttributeTest',
-				'atomicType',
-				'anyItemType',
-				'parenthesizedItemType',
-				'typedMapTest',
-				'typedArrayTest',
-				'nameTest',
-				'Wildcard',
-			]);
-
-			const testExpression = compileTest(test, disallowUpdating(compilationOptions));
+			const tests = astHelper
+				.getChildren(step, '*')
+				.filter((test) =>
+					[
+						'attributeTest',
+						'anyElementTest',
+						'piTest',
+						'documentTest',
+						'elementTest',
+						'commentTest',
+						'namespaceTest',
+						'anyKindTest',
+						'textTest',
+						'anyFunctionTest',
+						'typedFunctionTest',
+						'schemaAttributeTest',
+						'atomicType',
+						'anyItemType',
+						'parenthesizedItemType',
+						'typedMapTest',
+						'typedArrayTest',
+						'nameTest',
+						'Wildcard',
+					].includes(test[0]),
+				);
+
+			const testExpressions = new CombinatingTest(
```

Comment:
nit: technically still a single (combined) test :)

Reply:
Correct. Reworked this a bit to actually parse stuff